src/module/item/spellcasting-entry/document.ts

```diff
@@ -370,6 +375,8 @@ class SpellcastingEntryPF2e<TParent extends ActorPF2e | null = ActorPF2e | null>
             throw ErrorPF2e("Spellcasting entries can only exist on characters and npcs");
         }
 
+        // Resolve spell references first
+        await this.references?.resolve();
```

Comment:
No rush on this due to remaster delays, but to leave a note: this should probably be moved to the collection's getSpellData().